servlet/_index.md

```diff
@@ -1,7 +1,173 @@
+Parameters can be sent to a servlet through the URI and they are processed via name-value pairs. Multiple values can be assigned to any parameter name.
+Once the parameters are passed, the servlet contains a number of methods that can be used for accessing the parameters.
+
+### Additional Features
+
+There are a number of other important features that `Jakarta Servlet` provides that go above and beyond the basics of displaying dynamic content and accepting request parameters.
+This section will provide a high level overview for a number of these features. Many of these additional features can be registered within the `web.xml` of an application, or they can be registered using annotations.
+
+`Servlet Filters` are lightweight and reusable Java components that allow on-the-fly modifications of header and payload for both the request and response.
+`Jakarta Servlet` provides a framework for writing filters and applying them to servlets. To write a filter, a class must extend `jakarta.servlet.Filter`, and implement the `doFilter()` method.
+The `@WebFilter` annotation can be applied to a class, along with metadata, to define it as a filter.
+
+`Servlet Listeners` can be used to listen for events and perform actions when the corresponding events occur. There are a number of different event types, from servlet startup/shutdown, to session or request events.
+Listener classes must implement a listener interface corresponding to the associated type of event listener being developed. Listener events include: `AsyncListener`, `ServletContextListener`,
+`ServletContextAttributeListener`, `ServletRequestListener`, `ServletRequestAttributeListener`, `HttpSessionListener`, `HttpSessionBindingListener`, `HttpSessionAttributeListener`, `HttpSessionActivationListener`.
+Listeners can be registered by applying the `@WebListener` annotation to a listener class.
+
+`Servlet containers` allow files to be uploaded when data is sent via `multipart/form-data`. To indicate that a servlet is used for performing upload, it must have the `@MultipartConfig` annotation applied to it.
+
+`Servlets` can access information pertaining to attributes and headers via provided APIs. Attributes are associated with each request, and they can be set to specify information.
+Headers are also associated with every HTTP request, and they can be used to specify information about sessions and caching, amongst other things.
+
+`Servlets` can make use of the URI path to invoke different resources. The path portion of a URI identifies which resource should be processed.
+Part of the `Jakarta Servlet` specification outlines a standard set of guidelines that should be followed by compliant containers when processing URI paths.
+A servlet request path conforms to the following format:
+
+```Java
+requestURI = contextPath + servletPath + pathInfo
+```
```

Comment:
Reconstructing a Request URI is 

* scheme (from combination of configuration, connectors, and request details)
* host / authority (from combination of configuration, connectors, request details, forwarding headers, etc)
* context path (with path params merged from original submitted request path)
* servlet path (with path params merged from original submitted request path)
* path info (optional)
* query (optional)

This get more complicated when RequestDispatcher is used (with regards to merging of original request details and request dispatcher details)

Reply:
This description also doesn't handle the nuance of how uri encoding / uri decoding / uri canonicalization impacts the resulting requestURI.
This is also a section that's best handled by pointing to the spec documentation.

Reply:
Yet another +1 and I'll add SNI and virtual hosting to the list of things that can impact mapping.